src/web/app/components/question-submission-form/question-submission-form.component.ts

```diff
@@ -90,18 +90,18 @@ export class QuestionSubmissionFormComponent implements OnInit {
   saveCommentEvent: EventEmitter<number> = new EventEmitter();
 
   visibilityStateMachine: VisibilityStateMachine;
-  allowedToHaveParticipantComment: boolean;
+  allowedToHaveParticipantComment: boolean = false;
 
   constructor(private feedbackQuestionsService: FeedbackQuestionsService,
               private feedbackResponseService: FeedbackResponsesService) {
     this.visibilityStateMachine =
         this.feedbackQuestionsService.getNewVisibilityStateMachine(
             this.model.giverType, this.model.recipientType);
-    this.allowedToHaveParticipantComment =
-        this.feedbackQuestionsService.isAllowedToHaveParticipantComment(this.model.questionType);
   }
 
   ngOnInit(): void {
+    this.allowedToHaveParticipantComment =
```

Comment:
I think it is better to put this line to around L45 (i.e. Every time `model` gets changed, the boolean value should also be changed).